values.go

```diff
@@ -38,6 +39,12 @@ func GetFlags(dir string) (map[string]string, error) {
 
 	if version, err := versionFromFile(dir); err != nil {
 		return nil, fmt.Errorf("failed to get version: %v", err)
+	} else if version == "" {
+		// ex: v0.1.2 0.1.2 0.1.2.dev3
+		matched, err := regexp.MatchString(`^v?(\d+\.){1,3}(\d+)`, gitSummary)
```

Comment:
untags version should return nothing actually，should'nt it? version should contains version, that  Ithink发自网易邮箱大师
            On 08/18/2016 11:32, Ahmet Alp Balkan wrote:In values.go:
> @@ -38,6 +39,12 @@ func GetFlags(dir string) (map[string]string, error) {
>   
>  	if version, err := versionFromFile(dir); err != nil {
>  		return nil, fmt.Errorf("failed to get version: %v", err)
> +	} else if version == "" {
> +		// ex: v0.1.2 0.1.2 0.1.2.dev3
> +		matched, err := regexp.MatchString(`^v?(\d+\.){1,3}(\d+)`, gitSummary)

this is still not ok. I'm not gonna accept a version regexp here. This should work on untagged repos as well, therefore please just call repo.Summary() and nothing else here.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/ahmetalpbalkan/govvv","title":"ahmetalpbalkan/govvv","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/ahmetalpbalkan/govvv"}},"updates":{"snippets":[{"icon":"PERSON","message":"@ahmetalpbalkan in #5: this is still not ok. I'm not gonna accept a version regexp here. This should work on untagged repos as well, therefore please just call repo.Summary() and nothing else here."}],"action":{"name":"View Pull Request","url":"https://github.com/ahmetalpbalkan/govvv/pull/5/files/54bb1f0fe19ea63771fc2026e885c97c8f2db5d0#r75244240"}}}